sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlParser.scala

```diff
@@ -327,9 +327,9 @@ class StaxXmlParser(
 
     if (valuesMap.isEmpty) {
       // Return an empty row with all nested elements by the schema set to null.
-      InternalRow.fromSeq(Seq.fill(schema.fieldNames.length)(null))
+      new GenericInternalRow(Array.fill[Any](schema.length)(null))
```

Comment:
`schema.fieldNames.length` => `schema.fields.map(_.name).length`

`schema.length` => `fields.length`

`schema.length` and `schema.fieldNames.length` should be equal, and `schema.length` can save a map conversion, so here it is casually changed to `schema.length`